app/code/Magento/MediaContentApi/Api/AssignAssetInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MediaContentApi\Api;
+
+/**
+ * Saving data represents relation between the media asset and media content
+ * @api
+ */
+interface AssignAssetInterface
+{
+    /**
+     * @param int $assetId
+     * @param string $contentType
+     * @param string $contentEntityId
+     * @param string $contentField
+     *
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
+     */
+    public function execute(int $assetId, string $contentType, string $contentEntityId, string $contentField): void;
```

Comment:
`content entity id` is an id of an entity which has content where an asset is used. For example, if we store a relation between asset and content of the product entity the `catalog_product_entity.entity_id` will be used for this. 
In a provided implementation this is used here: https://github.com/magento/magento2/pull/27536/files#diff-425fc7fd56cadd0cbbf4f2b600c74771R107

The implementation doesn't limit supported types of object with content to provide extensibility. From the box, the `cms_page`, `cms_block`, `catalog_product`, `catalog_category` provided as a basic content type. If someone will need to store information about media asset usage in the content of a custom object it will create its own content type and this automatically be used for all other features such as showing where and how an asset is used.

Reply:
@buskamuza we've introduced ContentIdentityInterface, please see the updated code